modules/dhcp_common/isc/configuration_parser.rb

```diff
@@ -427,11 +427,13 @@ def visit_parse_tree_nodes(parse_tree, all_hosts, all_subnets, all_ignored, root
         end
 
         def parse_file(a_path)
+          a_path = @@config_path + '/' + a_path if File.dirname(a_path) == '.'
```

Comment:
This should be:

    a_path = File.join(@config_path, a_path) if File.dirname(a_path) == '.'

Also I am not so sure about the test you perform here. Dot is not the only case, it can be also path like `../dhcp/some_file` which will lead to `..` that does not match. I suggest to convert the path to absolute path: https://ruby-doc.org/core-2.1.5/File.html#method-c-absolute_path

This would be (untested):

    absolute_path = File.absolute_path(a_path, @config_path)

This will resolve correctly for both absolute paths and relative paths.